### PR TITLE
Regen

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin @sdroege

### CODE REVIEW

**Comment by EPashkin** [comment]
👍

File: src/auto/functions.rs [hunk 1]
```diff
@@ -47,9 +47,7 @@ pub fn assertion_message_cmpstr(domain: &str, file: &str, line: i32, func: &str,
     }
 }
 
-pub fn assertion_message_expr<'a, 'b, P: Into<Option<&'a str>>, Q: Into<Option<&'b str>>>(domain: P, file: &str, line: i32, func: &str, expr: Q) {
-    let domain = domain.into();
-    let expr = expr.into();
+pub fn assertion_message_expr(domain: Option<&str>, file: &str, line: i32, func: &str, expr: Option<&str>) {
```
Comment (sdroege) [inline src/auto/functions.rs]: Why do we even have this function in the bindings? Someone should review all the global functions we have generated here...

File: src/auto/functions.rs [hunk 2]
```diff
@@ -1218,8 +1208,7 @@ pub fn test_set_nonfatal_assertions() {
     }
 }
 
-pub fn test_skip<'a, P: Into<Option<&'a str>>>(msg: P) {
-    let msg = msg.into();
+pub fn test_skip(msg: Option<&str>) {
```
Comment (sdroege) [inline src/auto/functions.rs]: Same for this for example. We generate lots of functions that are useless in Rust, not usable as-is from Rust, or are simply wrong :)

File: src/value_array.rs [hunk 3]
```diff
@@ -28,8 +28,7 @@ impl ValueArray {
         }
     }
 
-    pub fn append<'a, P: Into<Option<&'a Value>>>(&mut self, value: P) {
-        let value = value.into();
+    pub fn append(&mut self, value: Option<&Value>) {
```
Comment (sdroege) [inline src/value_array.rs]: This should not be `nullable`. While the C API allows it, it is unsafe. Same for the other functions below

**Comment by sdroege** [comment]
Looks good to me apart from the value array stuff. And please create an issue for reviewing the global functions here and removing most of them.

**Comment by GuillaumeGomez** [comment]
I removed the two useless functions you pointed and removed the `Option` from the array function.

File: src/value_array.rs [hunk 4]
```diff
@@ -28,7 +28,7 @@ impl ValueArray {
         }
     }
 
-    pub fn append(&mut self, value: Option<&Value>) {
+    pub fn append(&mut self, value: &Value) {
```
Comment (sdroege) [inline src/value_array.rs]: Also for `insert` and `prepend` below

File: Gir.toml [hunk 5]
```diff
@@ -93,6 +93,12 @@ status = "generate"
     #Quark
     ignore = true
     [[object.function]]
+    name = "assertion_message_expr"
+    ignore = true # function is useless
+    [[object.function]]
+    name = "test_skip"
```
Comment (sdroege) [inline Gir.toml]: You could make this `pattern = "test_*"` I guess. The GLib unit tests API

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by sdroege** [comment]
:+1:

File: src/source.rs [hunk 6]
```diff
@@ -270,7 +270,7 @@ where F: FnMut() -> Continue + 'static {
 /// process exits.
 ///
 /// `func` will be called when `pid` exits
-pub fn child_watch_add<'a, N: Into<Option<&'a str>>, F>(pid: Pid, func: F) -> SourceId
+pub fn child_watch_add<F>(pid: Pid, func: F) -> SourceId
```
Comment (EPashkin) [inline src/source.rs]: Strange that there was unused constraint N but rust don't nag about it

**Comment by EPashkin** [comment]
👍 after `insert` and `prepend` has option

**Comment by GuillaumeGomez** [comment]
> 👍 after `insert` and `prepend` has option

I removed them on purpose after @sdroege asked for it so I'll keep it this way.

@sdroege: It's ready on my side so if there's nothing more missing, please merge. :)

**Comment by sdroege** [comment]
> +1 after `insert` and `prepend` has option

@EPashkin what do you mean?

**Comment by EPashkin** [comment]
Sorry, I made mistake Option need be removed and this already done.

**Comment by EPashkin** [comment]
Lets merge?

**Comment by GuillaumeGomez** [comment]
Yup!

**Pull request merged**